train.py

```diff
@@ -548,7 +553,7 @@ def get_optimizer(model_parameters):
 
     load_optimizer_states = config.get('load_optimizer_states', True)
     # if resuming and not loading optimizer states, we can't use warmup or the LR never changes from the initial value (still don't know why)
-    if 'warmup_steps' in config and config['warmup_steps'] > 0 and load_optimizer_states:
+    if warmup_steps > 0 and load_optimizer_states:
         warmup_steps = config['warmup_steps']
```

Comment:
I think you want to delete the line here, as it overrides the new value with the one with fractions.

Reply:
Nice catch!

It was a copy and paste fail from [here](https://github.com/jukofyork/qlora-pipe/blob/decoupled-orthogonal-regularisation/train.py) file trying to use GitHub online editor :frowning: